src/client/dfs/dfs.c

```diff
@@ -2012,13 +2012,13 @@ dfs_readdir(dfs_t *dfs, dfs_obj_t *obj, daos_anchor_t *anchor, uint32_t *nr,
 	if (rc)
 		return rc;
 
-	D_ALLOC_ARRAY(kds, *nr);
+	D_MM_ALLOC_ARRAY(kds, *nr);
```

Comment:
A better way to improve performance of this function would be to simple save kds and enum_buf between calls by hanging them on the dfs_obj_t somewhere.  I think performance orientated users would likely call dfs_iterate() over dfs_readdir() so this function wouldn't be high on my list.

Reply:
Good idea. Probably I can do this, but in separate PR.